packages/cryptoplease/lib/features/payment_request/src/bl/repository.dart

```diff
@@ -49,6 +49,10 @@ class PaymentRequestRepository {
   Future<void> save(PaymentRequest payment) =>
       _db.into(_db.paymentRequestRows).insertOnConflictUpdate(payment.toRow());
 
+  Future<void> delete(PaymentRequest payment) =>
+      (_db.delete(_db.paymentRequestRows)..whereSamePrimaryKey(payment.toRow()))
```

Comment:
Doesn't make sense to convert the model to a row just for comparing keys later. You can filter by the primary key directly.